tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3134,4 +3134,35 @@ public function publicDownloadsTheFolderFromTheLastCreatedPublicLink(string $res
 			)
 		);
 	}
+
+	/**
+	 * @Then /^the json responded should contain these key and value pairs:$/
+	 *
+	 * Loops through data items in response to find given key and value pairs
+	 * It does not support field value replacing
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theJsonRespondedShouldContainTheseKeyAndValuePairs(TableNode $table):void {
+		$this->featureContext->verifyTableNodeColumns($table, ['key', 'value']);
+		$response = json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
+		$data = $response["ocs"]["data"];
+		foreach ($table->getHash() as $row) {
+			$key = $row['key'];
+			$expectedValue = $row['value'];
+			$segments = explode('@@@', $key);
+			$actualValue  = null;
+			$tempArray =  $data;
+			foreach ($segments as $segment) {
+				$arrayKeyExists = \array_key_exists($segment, $tempArray);
+				Assert::assertTrue($arrayKeyExists, "The key $segment does not exist on the response");
+				$actualValue = $tempArray[$segment];
+				$tempArray = $actualValue;
+			}
+			Assert::assertEquals($expectedValue, $actualValue, "Expected $expectedValue but received $actualValue");
+		}
```

Comment:
```suggestion
			$actualValue  = $data;
			foreach ($segments as $segment) {
				$arrayKeyExists = \array_key_exists($segment, $tempArray);
				Assert::assertTrue($arrayKeyExists, "The key $segment does not exist on the response");
				$actualValue = $actualValue[$segment];
			}
			Assert::assertEquals($expectedValue, $actualValue, "Expected $expectedValue but received $actualValue");
		}
```
can you check if like this works for the loop

Reply:
Works.